File: SAMPLES/SWIFT/classes.swift
// classes.swift - Classes compilees par SWIFT86
// Demontre : class, heritage, override, is, as?, vtable

class Animal {
    var name: String
    var legs: Int

    func speak() -> Int {
        return 0
    }
}

class Dog: Animal {
    var breed: String

    override func speak() -> Int {
        return 1
    }
}

class Cat: Animal {
    var indoor: Bool

    override func speak() -> Int {
        return 2
    }
}

let pet: Animal = Animal()
print(42)

File: SAMPLES/CPP/classes.cpp
/* classes.cpp - class, heritage, methodes virtuelles */

class Animal {
public:
    int age;
    Animal() {
        age = 0;
    }
    virtual int speak() {
        return 1;
    }
};

class Dog : public Animal {
public:
    int speak() {
        return 2;
    }
};

int main() {
    Dog d;
    d.age = 5;
    int s = d.speak();
    cout << d.age << endl;
    cout << s << endl;
    return 0;
}
